### PR TITLE
[KOGITO-1347] Test Data index using external Kafka

### PR DESCRIPTION
**Pull request by sutaakar**
https://issues.redhat.com/browse/KOGITO-1347

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by sutaakar** [comment]
Based on https://github.com/kiegroup/kogito-cloud-operator/pull/315 (waiting for https://github.com/kiegroup/kogito-cloud-operator/pull/315 to be merged first)

**Comment by sutaakar** [comment]
Can be used as a reproducer for https://issues.redhat.com/browse/KOGITO-2012

**Comment by sutaakar** [comment]
@MarianMacik @jakubschwan Can you please take a look?

File: test/features/install_dataindex.feature [hunk 1]
```diff
@@ -41,3 +41,28 @@ Feature: Kogito Data Index
...
     }
     """
+
+
+#####
+
+  # Used just to confirm that integration between Data index and external Kafka works.
+  # Will be removed once external Kafka integration is implemented for Kogito service.
+  # Kogito service scenario will cover both Kogito service and Data index integration with external Kafka.
+  # Kafka connection using "instance" parameter is blocked due to https://issues.redhat.com/browse/KOGITO-2012, will be implemented once the JIRA is resolved.
```
Comment (MarianMacik) [inline test/features/install_dataindex.feature:51]: JIRA is resolved now.

File: go.sum [hunk 2]
```diff
@@ -63,6 +63,7 @@ github.com/Microsoft/go-winio v0.4.12/go.mod h1:VhR8bwka0BXejwEJY73c50VrPtXAaKcy
 github.com/Microsoft/hcsshim v0.0.0-20190417211021-672e52e9209d/go.mod h1:Op3hHsoHPAvb6lceZHDtd9OkTew38wNoXnJs8iY7rUg=
 github.com/Microsoft/hcsshim v0.8.6/go.mod h1:Op3hHsoHPAvb6lceZHDtd9OkTew38wNoXnJs8iY7rUg=
 github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46/go.mod h1:3wb06e3pkSAbeQ52E9H9iFoQsEEwGN64994WTCIhntQ=
+github.com/NYTimes/gziphandler v1.0.1 h1:iLrQrdwjDd52kHDA5op2UBJFjmOb9g+7scBan4RN8F0=
```
Comment (MarianMacik) [inline go.sum:66]: Curious why this was changed when you didn't add any new Go modules.

Comment (sutaakar) [reply go.sum:66]: Good question.

**Comment by sutaakar** [comment]
Rebased and adjusted.
@radtriste Can you please take a final looks and merge if it is ok?

File: test/features/install_dataindex.feature [hunk 3]
```diff
@@ -41,3 +41,32 @@ Feature: Kogito Data Index
...
+  # Used just to confirm that integration between Data index and external Kafka works.
+  # Will be removed once external Kafka integration is implemented for Kogito service.
+  # Kogito service scenario will cover both Kogito service and Data index integration with external Kafka.
+  @externalcomponent
+  @kafka
+  Scenario Outline: Install Kogito Data Index using external Kafka
+    Given Kafka instance "external-kafka" is deployed
+
+    When Install Kogito Data Index with 1 replicas with configuration:
+      | kafka | <kafka-second-column-name> | <kafka-second-column-value> |
```
Comment (radtriste) [inline test/features/install_dataindex.feature:57]: `<kafka-key>` & `<kafka-value>` ?

Comment (sutaakar) [reply test/features/install_dataindex.feature]: updated

**Comment by radtriste** [comment]
CircleCi is success but not reported here ...
https://circleci.com/gh/kiegroup/kogito-cloud-operator/1620?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Pull request merged**